framework/src/modules/interoperability/mainchain/module.ts

```diff
@@ -299,60 +288,65 @@ export class MainchainInteroperabilityModule extends BaseInteroperabilityModule
-			}
-
-			// chainInfos should be ordered lexicographically by chainInfo.chainID
-			const sortedByChainID = [...chainInfos].sort((a, b) => a.chainID.compare(b.chainID));
-			for (let i = 0; i < chainInfos.length; i += 1) {
-				if (!chainInfos[i].chainID.equals(sortedByChainID[i].chainID)) {
-					throw new Error('chainInfos is not ordered lexicographically by chainID.');
-				}
-			}
-
-			// The entries chainData.name must be pairwise distinct
-			const chainDataNames = chainInfos.map(info => info.chainData.name);
-			if (new Set(chainDataNames).size !== chainDataNames.length) {
-				throw new Error(`chainData.name must be pairwise distinct.`);
-			}
-
 			this._verifyChainInfos(ctx, chainInfos);
 			this._verifyTerminatedStateAccounts(chainInfos, terminatedStateAccounts);
 			this._verifyTerminatedOutboxAccounts(
 				chainInfos,
 				terminatedStateAccounts,
 				terminatedOutboxAccounts,
 			);
+
+			await this.processGenesisState(ctx, genesisInteroperability);
 		}
 	}
 
 	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
 	private _verifyChainInfos(ctx: GenesisBlockExecuteContext, chainInfos: ChainInfo[]) {
-		const mainchainID = getMainchainID(ctx.chainID);
-
-		// verify root level properties
-		for (const chainInfo of chainInfos) {
-			const { chainID } = chainInfo;
+		// Each entry chainInfo in chainInfos has a unique chainInfo.chainID
+		const chainIDs = chainInfos.map(info => info.chainID);
+		if (!objectUtils.bufferArrayUniqueItems(chainIDs)) {
+			throw new Error(`chainInfos doesn't hold unique chainID.`);
+		}
 
-			// chainInfo.chainID != getMainchainID();
-			if (chainID.equals(mainchainID)) {
-				throw new Error(`chainID must be not equal to ${mainchainID.toString('hex')}.`);
+		// chainInfos should be ordered lexicographically by chainInfo.chainID
+		const sortedByChainID = [...chainInfos].sort((a, b) => a.chainID.compare(b.chainID));
+		for (let i = 0; i < chainInfos.length; i += 1) {
+			if (!chainInfos[i].chainID.equals(sortedByChainID[i].chainID)) {
+				throw new Error('chainInfos is not ordered lexicographically by chainID.');
 			}
+		}
 
-			// - chainInfo.chainId[0] == getMainchainID()[0].
-			if (chainID[0] !== mainchainID[0]) {
-				throw new Error(`chainID[0] doesn't match ${mainchainID[0]}.`);
-			}
+		// The entries chainData.name must be pairwise distinct
+		const chainDataNames = chainInfos.map(info => info.chainData.name);
+		if (new Set(chainDataNames).size !== chainDataNames.length) {
+			throw new Error(`chainData.name must be pairwise distinct.`);
+		}
 
+		const mainchainID = getMainchainID(ctx.chainID);
+
+		// verify root level properties
+		for (const chainInfo of chainInfos) {
+			this._verifyChainID(chainInfo, mainchainID);
 			this._verifyChainData(ctx, chainInfo);
 			this._verifyChannelData(ctx, chainInfo);
 			this._verifyChainValidators(chainInfo);
 		}
 	}
 
+	private _verifyChainID(chainInfo: ChainInfo, mainchainID: Buffer) {
```

Comment:
I think this function could be moved to the base_module file and should be called also in `verifyTerminatedStateAccountsCommon`, given the last LIP update (as discussed also in two previous comments -- [here](https://github.com/LiskHQ/lisk-sdk/pull/8317#discussion_r1159580452) and [here](https://github.com/LiskHQ/lisk-sdk/pull/8317#discussion_r1159576501))